test/test_resume.cpp

```diff
@@ -1690,3 +1690,42 @@ TORRENT_TEST(resume_data_have_pieces)
 	TEST_EQUAL(rs->params.unfinished_pieces.size(), 1);
 }
 
+// See https://github.com/arvidn/libtorrent/issues/5174
+TORRENT_TEST(removed)
+{
+	// we're _likely_ to trigger the condition, but not guaranteed. loop
+	// until we do.
+	while (true) {
```

Comment:
it's risky to have an infinite loop in a test. What if you run this on a machine that never triggers the condition? Or under a build configuration that doesn't use a separate thread for disk I/O. Then everything is deterministic, and it may not trigger this condition.

Reply:
In such a configuration, does the libtorrent network thread still exist? Are the `async_call()`s actually synchronous?

Reply:
the network thread exists, but no other thread (this is the case when building with posix disk I/O in 2.0). So anything that may be asynchronous/non-deterministic because of the disk I/O will be deterministic in that configuration.

Reply:
I actually thought the condition would still get triggered because the `torrent` is kept alive by a pointer somewhere in the session's `io_service`.

If the sequence _doesn't_ trigger the condition (`save_resume_data()` doesn't generate any alerts), how will the caller get notified of this? Didn't it already successfully submit the call?

Reply:
I would expect that any call to save_resume_data() would result in exactly one response alert or an exception. Still, if there's a bug, it would be better with an explicit test failure than a hung test (that won't produce any output on CI)